include/crocoddyl/core/control-base.hpp

```diff
@@ -0,0 +1,187 @@
+                        const Eigen::Ref<const VectorXs>& p) const = 0;
+
+  // virtual MatrixXs calcDiff_J(double t, const Eigen::Ref<const VectorXs>& p) const;
+
+  /**
+   * @brief Compute the product between a specified matrix and the Jacobian of the control (with respect to the
+   * parameters)
+   *
+   * @param[in]  t      Time
+   * @param[in]  p      Control parameters
+   * @param[in]  A      A matrix to multiply times the Jacobian
+   * @param[out] out    Product between the matrix A and the Jacobian of the control with respect to the parameters
+   */
+  virtual void multiplyByJacobian(double t, const Eigen::Ref<const VectorXs>& p, const Eigen::Ref<const MatrixXs>& A,
+                                  Eigen::Ref<MatrixXs> out) const = 0;
+
+  virtual MatrixXs multiplyByJacobian_J(double t, const Eigen::Ref<const VectorXs>& p,
+                                        const Eigen::Ref<const MatrixXs>& A) const;
+
+  /**
+   * @brief Compute the product between the transposed Jacobian of the control (with respect to the parameters) and
+   * a specified matrix
+   *
+   * @param[in]  t      Time
+   * @param[in]  p      Control parameters
+   * @param[in]  A      A matrix to multiply times the Jacobian
+   * @param[out] out    Product between the transposed Jacobian of the control with respect to the parameters and the
+   * matrix A
+   */
+  virtual void multiplyJacobianTransposeBy(double t, const Eigen::Ref<const VectorXs>& p,
+                                           const Eigen::Ref<const MatrixXs>& A, Eigen::Ref<MatrixXs> out) const = 0;
+
+  virtual MatrixXs multiplyJacobianTransposeBy_J(double t, const Eigen::Ref<const VectorXs>& p,
+                                                 const Eigen::Ref<const MatrixXs>& A) const;
+  /**
+   * @brief Return the dimension of the control value
+   */
+  std::size_t get_nu() const;
+
+  /**
+   * @brief Return the dimension of control parameters
+   */
+  std::size_t get_np() const;
+
+ protected:
+  std::size_t nu_;  //!< Control dimension
+  std::size_t np_;  //!< Control parameters dimension
+};
+
+template <typename _Scalar>
+struct ControlParametrizationDataAbstractTpl {
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+
+  template <template <typename Scalar> class Model>
+  explicit ControlParametrizationDataAbstractTpl(Model<Scalar>* const model)
+      : u_diff(model->get_nu()), u_params(model->get_np()), J(model->get_nu(), model->get_np()) {
+    u_diff.setZero();
+    u_params.setZero();
+    J.setZero();
+  }
+  virtual ~ControlParametrizationDataAbstractTpl() {}
+
+  VectorXs u_diff;    //!< value of the control
+  VectorXs u_params;  //!< value of the control parameters
+  MatrixXs J;         //!< Jacobian of the control
```

Comment:
Yes, the Jacobian at the moment is only used for debugging. Providing a default implementattion fo the other two metthods is a good idea, but not straightforward to implement because calcDiff takes the data as input, whereas these two metthods don't, so I should allocate a temporary Data structure inside the methhod, which seems a waste to me. Any idea for a workaround?